services/QuillLMS/app/models/activity_session.rb

```diff
@@ -105,6 +105,7 @@ class StudentNotAssignedActivityError < StandardError; end
   # FIXME: do we need the below? if we omit it, may make things faster
   default_scope -> { joins(:activity) }
 
+  scope :finished, -> { where(state: STATE_FINISHED) }
   scope :completed,  -> { where.not(completed_at: nil) }
```

Comment:
I don't know that this is wrong, but `finished` and `completed` scopes should be identical.  If they aren't that's some sort of bug.  So it may make sense to settle on one to be the "canonical" scope to use rather than having some places in the code use one and other places use the other.